tensorbay/client/tests/test_dataset.py

```diff
@@ -145,6 +151,48 @@ def test__move_segment(self, mocker):
             "POST", "segments?move", self.dataset_client._dataset_id, json=post_data
         )
 
+    def test_enable_cache(self, mocker, mock_list_segments, mock_list_data_details, tmp_path):
+        self.dataset_client._status.checkout(draft_number=1)
+        with pytest.raises(StatusError):
+            self.dataset_client.enable_cache()
+
+        self.dataset_client._status.checkout(commit_id="commit-1")
+        cache_path = tmp_path / "cache_data"
+        dataset_cache_path = cache_path / self.dataset_client.dataset_id
+        self.dataset_client.enable_cache(cache_path)
+        assert self.dataset_client.cache_enabled == True
+        assert self.dataset_client._cache_path == str(dataset_cache_path)
+
+        _, list_segments = mock_list_segments(mocker)
+        segment_name = list_segments["segments"][0]["name"]
+        segment = self.dataset_client.get_segment(segment_name)
+        mock_list_data_details(mocker)
+        segment_data = segment.list_data()
+        urlopen = mocker.patch(
+            f"{file.__name__}.RemoteFileMixin._urlopen",
+            return_value=mock_response(read=lambda *args: bytes(1)),
+        )
+
+        segment_cache_path = (
+            dataset_cache_path / self.dataset_client.status.commit_id / segment_name
+        )
+        for i in range(3):  # Traverse the segment for 3 times.
```

Comment:
will it be better to separate this loop into two? one for the first iter, then the next two iter?